src/main/java/nextstep/subway/domain/SectionLineUp.java

```diff
@@ -0,0 +1,171 @@
+package nextstep.subway.domain;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * 노선 구간 목록을 관리하는 일급 컬렉션
+ */
+@Embeddable
+public class SectionLineUp {
+
+    private static final int START_INDEX = 0;
+
+    @OneToMany(mappedBy = "line", orphanRemoval = true, cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    private List<Section> sectionList = new ArrayList<>();
+
+    protected SectionLineUp() {
+    }
+
+    public void addSection(Section section) {
+        if (sectionList.isEmpty()) {
+            sectionList.add(section);
+            return;
+        }
+        validUnknownStation(section);
+        validSameSection(section);
+        add(section);
+    }
+
+    private void add(Section section) {
+        if (!isEndUpStation(section) && !isEndDownStation(section)) {
+            createInternal(section);
+            return;
+        }
+        sectionList.add(section);
+    }
+
+    public List<Section> getSectionsInOrder() {
+        Section firstSection = findFirstSection(sectionList.get(START_INDEX));
+        return searchSectionsByOrder(new ArrayList<>(), firstSection);
+    }
+
+    public Stations getStationsInOrder() {
+
+        Section firstSection = findFirstSection(sectionList.get(START_INDEX));
+        final List<Station> stations = searchStationsByOrder(new ArrayList<>(), firstSection);
+        return new Stations(stations.stream().distinct().collect(Collectors.toList()));
+    }
+
+    private boolean isEndUpStation(Section section) {
+        //  출발지가 같은 노선이 없고, 도착지에서 출발하는 노선은 있지만, 도착지로 향하는 노선이 없는 경우
+        return notHasSameUpStation(section) && notHasSameDownStation(section) && hasSameUpStationByDownStation(section);
+    }
```

Comment:
```suggestion
    private boolean isEndUpStation(Section section) {
        //  출발지가 같은 노선이 없고, 도착지에서 출발하는 노선은 있지만, 도착지로 향하는 노선이 없는 경우
        return notHasSameDownStation(section) && hasSameUpStationByDownStation(section);
    }
```
`hasSameUpStationByDownStation`가 true면서 상행선이 같은 경우는
상행선과 하행선이 모두 존재한다는 의미기 때문에 `notHasSameUpStation`는 체크할 필요가 없을거같네요. 😀
상행선과 하행선이 모두 존재하는 경우는 유효성검사에서 미리 걸러냈으니깐요!

Reply:
다른 조건에서 검증이 완료 된 케이스를 다시 검증하고 있었네요 :cry:
피드백 반영하여 이미 검증 안료 된 케이스 제거하였습니다 !